EventFilter/CastorRawToDigi/src/ZdcUnpacker.cc

```diff
@@ -0,0 +1,520 @@
+                  if ( id1.channel() > 12 ) dotp = false;
+                  if ( dotp ) {
+                  // std::cout << " tp_first="<< tp_first << " tp_last="<< tp_last<< " tb="<<htr_tb<<" slot="<<htr_slot<<" crate="<<htr_cr<<" dccid="<< dccid<< std::endl;
+                  // regular TPs (not HO)
+                  for (tp_work=tp_begin; tp_work!=tp_end; tp_work++) {
+                  //      std::cout << "raw=0x"<<std::hex<< tp_work->raw()<<std::dec <<std::endl;
+                  if (tp_work->raw()==0xFFFF) continue; // filler word
+                  if (slbAndChan(*tp_work)!=currFiberChan) { // start new set
+                  npre=0;
+                  currFiberChan=slbAndChan(*tp_work);
+
+                  // std::cout<< " NEW SET "<<std::endl;
+                  //HcalElectronicsId eid(tp_work->slbChan(),tp_work->slb(),spigot,dccid,htr_cr,htr_slot,htr_tb);
+                  //DetId did=emap.lookupTrigger(eid);
+                  //if (did.null()) {
+                  //report.countUnmappedTPDigi(eid);
+                  //if (unknownIdsTrig_.find(eid)==unknownIdsTrig_.end()) {
+                  //if (!silent) edm::LogWarning("HCAL") << "HcalUnpacker: No trigger primitive match found for electronics id :" << eid;
+                  //unknownIdsTrig_.insert(eid);
+                  //}
+                  //valid=false;
+                  //continue;
+                  //} else if (did==HcalTrigTowerDetId::Undefined ||
+                  //(did.det()==DetId::Hcal && did.subdetId()==0)) {
+                  //// known to be unmapped
+                  //valid=false;
+                  //continue;
+                  //}
+
+                  //colls.tpCont->push_back(HcalTriggerPrimitiveDigi(id));
+                  // set the various bits
+                  //if (!tpgSOIbitInUse) colls.tpCont->back().setPresamples(nps);
+                  //colls.tpCont->back().setZSInfo(htr.isUnsuppressed(),htr.wasMarkAndPassZSTP(slb(*tp_work),slbChan(*tp_work)));
+
+                  // no hits recorded for current
+                  ncurr=0;
+                  valid=true;
+                  }
+                  // add the word (if within settings or recent firmware [recent firmware ignores startSample/endSample])
+                  if (valid && ((tpgSOIbitInUse && ncurr<10) || (ncurr>=startSample_ && ncurr<=endSample_))) {
+                  colls.tpCont->back().setSample(colls.tpCont->back().size(),*tp_work);
+                  colls.tpCont->back().setSize(colls.tpCont->back().size()+1);
+                  }
+                  // set presamples,if SOI
+                  if (valid && tpgSOIbitInUse && isTPGSOI(*tp_work)) {
+                  colls.tpCont->back().setPresamples(ncurr);
+                  }
+                  ncurr++;
+                  npre++;
+                  }
+                  }*/
+
+    //////////////////////////////////////////////
+    qie_begin=(HcalQIESample*)daq_first;
```

Comment:
this const cast away was flagged by the static code analyzer 
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12467/9790/llvm-analysis/
please resolve this